terracotta/drivers/raster_base.py

```diff
@@ -36,6 +38,49 @@
 logger = logging.getLogger(__name__)
 
 
+class LFUCacheWithCompression(LFUCache):
+    """ Least-frequently-used Cache with data compression
+    """
+
+    def __getitem__(self, key: Any) -> np.ma.MaskedArray:
+        compressed_item = super().__getitem__(key)
+        return self._decompress_tuple(compressed_item)
+
+    def __setitem__(self, key: Any, value: np.ma.MaskedArray) -> None:
+        super().__setitem__(key, self._compress_ma(value))
+
+    def _compress_ma(self,
+                     arr: np.ma.MaskedArray) -> Tuple[bytes,
```

Comment:
This hurts my sense of esthetics (and we don't do this anywhere else in the Terracotta codebase), so I think adding

```python
CompressedArray = Tuple[bytes, bytes, str, Tuple[int, int]]
```

somewhere in the header would be nicer and more expressive as to what you are doing.